src/lib.rs

```diff
@@ -35,6 +35,20 @@ use colored::*;
 use log::{Level, LevelFilter, Log, Metadata, Record, SetLoggerError};
 use std::collections::HashMap;
 
+macro_rules! LOGGING_FMT {
```

Comment:
Does this need to use `macro_rules!` here? (I really don't know, a comment explaining why might be useful).

Reply:
unfortunally yes. the format_args!() macros require a literal string to construct the code generating the output. Unlike C the format string itself isn't kept at runtime.

Fun fact: my first attempt was something like

    let integer = 1234;
    let formatted = format_args!("{}", integer);

    #[cfg(not(feature = "stderr"))]
    println!("{}", formatted);

    #[cfg(feature = "stderr")]
    eprintln!("{}", formatted);

Unfortunally Rust doesnt agree :/. I see if i can come up with something better, but it should stay simple.

Reply:
this is fine if it's needed, it's not a problem.

Reply:
This issue is really annoying. I'd be really happy about seeing a `format_args!` which I can keep around without lifetime issues one day. FWIW, here's how the code might look like without that macro: https://github.com/daboross/fern/pull/84/commits/c56f4653810709d31074ca1e313612aa9d4b493d#diff-88abb15fc70475a7435ddde2f48b280df03419137e4d5736ccdecf4d885c8b32R1361-R1370

Reply:
yes i was surprised and annoyed as well. Still works it's only cosmetics. Some day when this is improved in rust it can be integrated here.

I am more annoyed that i have to use a macro instead a 'const', i am wondering if this restriction could be relaxed.